ssl/t1_lib.c

```diff
@@ -3513,7 +3513,7 @@ static int tls_decrypt_ticket(SSL *s, const unsigned char *etick,
         int rv = tctx->tlsext_ticket_key_cb(s, nctick, nctick + 16,
                                             &ctx, &hctx, 0);
         if (rv < 0)
-            return -1;
+            goto err;
         if (rv == 0)
             return 2;
```

Comment:
As return value 0 is when key is not matched. In this case, is application expected to call HMAC_Init_ex() and EVP_DecryptInit_ex(). 

If there is no assumption on this, then yes, need to clean up here as well.